backend/src/controllers/user.js

```diff
@@ -394,4 +395,25 @@ export default {
       next(err);
     }
   },
+
+  /**
+   * Express.js controller for POST /users/choose-fav-deck/:id
+   * @param {import('express').Request} req
+   * @param {import('express').Response} res
+   * @param {import('express').NextFunction} next
+   * @returns {Promise<void>}
+   */
+  chooseFavDeck: async (req, res, next) => {
+    try {
+      const deck = await deckService.findById(req.params.id);
+      if (!deck) { return res.sendStatus(404); }
+      if (deck.userId !== req.user.id) { return res.sendStatus(403); }
+
+      const chooseFavDeck = await userService.chooseFavDeck(req.user, req.params.id);
+      res.json(chooseFavDeck);
```

Comment:
```suggestion
      const choosedFavDeck = await userService.chooseFavDeck(req.user, req.params.id);
      res.json(choosedFavDeck);
```

simple histoire de naming, histoire d'etre le plus précis possible sur le nom des variables